mephisto/abstractions/providers/mturk/mturk_utils.py

```diff
@@ -61,21 +61,55 @@ def check_aws_credentials(profile_name: str) -> bool:
 ) -> bool:
+    if not os.path.exists(os.path.expanduser("~/.aws/")):
+        os.makedirs(os.path.expanduser("~/.aws/"))
+    aws_credentials_file_path = "~/.aws/credentials"
+    expanded_aws_file_path = os.path.expanduser(aws_credentials_file_path)
     try:
         # Check existing credentials
         boto3.Session(profile_name=profile_name)
         if register_args is not None:
             # Eventually we could manually re-parse the file and see
             # if the credentials line up or not, then fix ourselves
-            print(
-                f"WARNING credentials provided, but there's already a "
-                f"profile for {profile_name}. If these don't line up, you'll "
-                f"need to manually navigate to your ~/.aws/credentials file "
-                f"and remove the entry for this profile name, then run again.\n"
-                f"As this profile is currently loading, we consider it "
-                f"successfully registered anyways."
-            )
+            aws_credentials_file_string = ""
+            with open(expanded_aws_file_path, "r") as aws_credentials_file:
+                aws_credentials_file_string = aws_credentials_file.read()
+            # accessing the aws_credentials_file
+            aws_credentials = aws_credentials_file_string.split("\n")
+            # iterating to get the profile
+            profileFound = False
+            for credentialIndex in range(0, len(aws_credentials)):
+                if profileFound:
+                    aws_access_key_id = aws_credentials[credentialIndex]
+                    aws_secret_access_key = aws_credentials[credentialIndex+1]
+                    break
+                if str(aws_credentials[credentialIndex]).startswith("[{}]".format(profile_name)):
+                    profileFound = True
+
+            if register_args.access_key_id != aws_access_key_id or register_args.secret_access_key != aws_secret_access_key:
+                with open(expanded_aws_file_path, "a+") as aws_credentials_file:
+                    # Clean up file
+                    if aws_credentials_file_string:
+                        if aws_credentials_file_string.endswith("\n\n"):
+                            pass
+                        elif aws_credentials_file_string.endswith("\n"):
+                            aws_credentials_file.write("\n")
+                        else:
+                            aws_credentials_file.write("\n\n")
+                    # overWrite login details
+                    aws_credentials_file.write("[{}]\n".format(profile_name))
+                    aws_credentials_file.write(
+                        "aws_access_key_id={}\n".format(
+                            register_args.access_key_id)
+                    )
+                    aws_credentials_file.write(
+                        "aws_secret_access_key={}\n".format(
+                            register_args.secret_access_key)
+                    )
```

Comment:
The write process actually simplifies if you're trying to follow your current strategy, as you can insert directly:
```
aws_credentials[credentialIndex+1] = ... # update to the new value
aws_credentials[credentialIndex+2] = ... # update to the new value

with open(expanded_aws_file_path, 'w') as aws_credentials_file:
  aws_credentials_file.write("\n".join(aws_credentials))
```
This ensures that the old values are no longer in the file, and have been replaced by the new ones.

You can also skip the `# clean up file` code as the file should already be prepared if you're only making edits.